koordinates/sets.py

```diff
@@ -74,27 +145,204 @@ def _deserialize(self, data, manager):
+
+    @is_bound
+    def create_draft_version(self):
+        """
+        Creates a new draft version from this model content.
+
+        If anything in the data object has changed then an import will begin immediately.
+        Otherwise to force a re-import from the previous sources call :py:meth:`koordinates.client.Client.start_import`.
+
+        :rtype: Client
+        :return: the new version
+        :raises Conflict: if there is already a draft version for this set.
+        """
+        target_url = self._client.get_url(
+            "SET_VERSION", "POST", "create", {"id": self.id}
+        )
+        r = self._client.request("POST", target_url, json={})
+        return self._manager.create_from_result(r.json())
+
+    @is_bound
+    def publish(self, version_id=None):
+        """
+        Creates a publish task just for this version, which publishes as soon as any import is complete.
+
+        :return: the publish task
+        :rtype: Publish
+        :raises Conflict: If the version is already published, or already has a publish job.
+        """
+        if not version_id:
+            version_id = self.version.id
+
+        target_url = self._client.get_url(
+            "SET_VERSION", "POST", "publish", {"id": self.id, "version_id": version_id},
+        )
+        r = self._client.request("POST", target_url, json={})
+        return self._client.get_manager(Publish).create_from_result(r.json())
+
+    @is_bound
+    def save(self, with_data=False):
+        """
+        Edits this draft layerversion.
+        # If anything in the data object has changed, cancel any existing import and start a new one.
+
+        :param bool with_data: if ``True``, send the data object, which will start a new import and cancel
+            any existing one. If ``False``, the data object will *not* be sent, and no import will start.
+        :raises NotAllowed: if the version is already published.
+        """
+        target_url = self._client.get_url(
+            "SET_VERSION",
+            "PUT",
+            "edit",
+            {"id": self.id, "version_id": self.version.id},
+        )
+        r = self._client.request(
+            "PUT", target_url, json=self._serialize(with_data=with_data)
+        )
         return self._deserialize(r.json(), self._manager)
+
+    @is_bound
+    def delete_version(self, version_id=None):
+        """
+        Deletes this draft version (revert to published)
+
+        :raises NotAllowed: if this version is already published.
+        :raises Conflict: if this version is already deleted.
+        """
+        if not version_id:
+            version_id = self.version.id
+
+        target_url = self._client.get_url(
+            "SET_VERSION",
+            "DELETE",
+            "single",
+            {"id": self.id, "version_id": version_id},
+        )
+        r = self._client.request("DELETE", target_url)
+        logger.info("delete_version(): %s", r.status_code)
```

Comment:
client.request already has logging for this stuff, so not sure this is necessary